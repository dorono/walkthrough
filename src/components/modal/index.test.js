import React from 'react';
import {shallow} from 'enzyme';
import {Modal, ModalFooter, ModalHeader, ModalBody} from './index';

const defaultProps = {
    setModalSize: false,
};

describe('Modal Component', () => {
    it('should render without errors', () => {
        shallow((
            <Modal>
                Test
            </Modal>
        ));
    });

    it('should have a css height value of "auto" upon rendering for the first time', () => {
        const wrapper = shallow(<Modal {...defaultProps}>Test</Modal>);
        const cssHeight = wrapper.instance().renderModalHeight();
        expect(cssHeight).toEqual(expect.objectContaining({
            height: 'auto',
        }));
    });

    it('should match snapshot', () => {
        expect(shallow(<Modal>Test</Modal>)).toMatchSnapshot();
    });
});

describe('ModalHeader Component', () => {
    it('should render without errors', () => {
        shallow((
            <ModalHeader>
                Test
            </ModalHeader>
        ));
    });

    it('should match snapshot', () => {
        expect(shallow(<ModalHeader />)).toMatchSnapshot();
    });
});

describe('ModalBody Component', () => {
    it('should render without errors', () => {
        shallow((
            <ModalBody>
                Test
            </ModalBody>
        ));
    });

    it('should match snapshot', () => {
        expect(shallow(<ModalBody />)).toMatchSnapshot();
    });
});

describe('ModalFooter Component', () => {
    it('should render without errors', () => {
        shallow((
            <ModalFooter>
                Test
            </ModalFooter>
        ));
    });

    it('should match snapshot', () => {
        expect(shallow(<ModalFooter />)).toMatchSnapshot();
    });
});
