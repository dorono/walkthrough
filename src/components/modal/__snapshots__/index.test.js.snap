// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should match snapshot 1`] = `""`;

exports[`ModalBody Component should match snapshot 1`] = `<div />`;

exports[`ModalFooter Component should match snapshot 1`] = `<div />`;

exports[`ModalHeader Component should match snapshot 1`] = `<div />`;
